app/src/lib/stores/app-store.ts

```diff
@@ -3153,4 +3151,69 @@ export class AppStore {
 
     return gitStore.addUpstreamRemoteIfNeeded()
   }
+
+  private forkPullRequestRemoteName(remoteName: string) {
+    return `${ForkedRemotePrefix}${remoteName}`
+  }
+
+  public async _checkoutPullRequest(
+    repository: Repository,
+    pullRequest: PullRequest
+  ): Promise<void> {
+    const gitHubRepository = forceUnwrap(
+      `Cannot checkout a PR if the repository doesn't have a GitHub repository`,
+      repository.gitHubRepository
+    )
+    const head = pullRequest.head
+    const isRefInThisRepo =
+      head.gitHubRepository &&
+      head.gitHubRepository.cloneURL === gitHubRepository.cloneURL
+
+    if (isRefInThisRepo) {
+      // We need to fetch FIRST because someone may have created a PR since the last fetch
+      await this._fetch(repository, FetchType.UserInitiatedTask)
```

Comment:
Does this need to be user initiated? I think the only real difference between the behaviour is error handling - we might want to block the subsequent checkout if this fails to get the new ref. Am I reading that right?

Reply:
Yea, that seems like a reasonable thing to do. Do you suggest making it a `BackgroundTask` task and wrapping it in a `try/catch` block?

Reply:
A `BackgroundTask` will not propagate the error up to the user, so a `try/catch` isn't necessary.

At this point I think there's two potential solutions:

 - use `UserInitiatedTask` and, if the fetch fails, an error gets shown and no checkout is attempted.

This is the safer option as it means you avoid the "missing ref" situation mentioned in the comment, but also ensures that you're working with the latest commit even if it's a known ref.

 - switch to `BackgroundTask` and, if the fetch fails, the code continues on and tries to checkout

This is a bit more risky, and means the actual issue (no network, so can't get new refs) is not actually mentioned in the error if it tries to checkout a missing ref.

I think we should just leave this as-is, and let it fail fast.